diff --git a/node_modules/expo-router/build/fork/NavigationContainer.js b/node_modules/expo-router/build/fork/NavigationContainer.js
index 3db5ea4..b81f7b9 100644
--- a/node_modules/expo-router/build/fork/NavigationContainer.js
+++ b/node_modules/expo-router/build/fork/NavigationContainer.js
@@ -41,11 +41,31 @@ function NavigationContainerInner({ direction = react_native_1.I18nManager.getCo
     (0, useDocumentTitle_1.useDocumentTitle)(refContainer, documentTitle);
     (0, imperative_api_1.useImperativeApiEmitter)(refContainer);
     const [lastUnhandledLink, setLastUnhandledLink] = react_1.default.useState();
+    const hasMounted = react_1.default.useRef(false);
+    const pendingUnhandledLink = react_1.default.useRef(null);
+    const onUnhandledLinking = react_1.default.useCallback((link) => {
+        if (hasMounted.current) {
+            setLastUnhandledLink(link);
+        }
+        else {
+            pendingUnhandledLink.current = { link };
+        }
+    }, []);
+    react_1.default.useEffect(() => {
+        hasMounted.current = true;
+        if (pendingUnhandledLink.current) {
+            setLastUnhandledLink(pendingUnhandledLink.current.link);
+            pendingUnhandledLink.current = null;
+        }
+        return () => {
+            hasMounted.current = false;
+        };
+    }, []);
     const { getInitialState } = (0, useLinking_1.useLinking)(refContainer, {
         enabled: isLinkingEnabled,
         prefixes: [],
         ...linking,
-    }, setLastUnhandledLink);
+    }, onUnhandledLinking);
     const linkingContext = react_1.default.useMemo(() => ({ options: linking }), [linking]);
     const unhandledLinkingContext = react_1.default.useMemo(() => ({ lastUnhandledLink, setLastUnhandledLink }), [lastUnhandledLink, setLastUnhandledLink]);
     const onReadyForLinkingHandling = (0, useLatestCallback_1.default)(() => {
